plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -124,12 +148,34 @@ public boolean equals(Object o)
     @Override
     public int hashCode()
     {
-        return Objects.hash(schemaName, tableName, tableType, snapshotId, unenforcedPredicate, enforcedPredicate);
+        return Objects.hash(schemaName, tableName, tableType, tableSchema, snapshotId, unenforcedPredicate, enforcedPredicate);
```

Comment:
this is not desirable since Schema doesn't implement equals/hashcode

Reply:
Maybe `tableSchema.asStruct()`? StructType implements equals/hashcode and seems to have all the important data from Schema, just not the cache Maps.